opentelemetry-sdk/src/opentelemetry/sdk/_metrics/metric_reader.py

```diff
@@ -12,6 +12,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+from abc import ABC, abstractmethod
+from typing import Callable, Iterable
 
-class MetricReader:
-    pass
+from typing_extensions import final
+
+from opentelemetry.sdk._metrics.point import AggregationTemporality, Metric
+
+_logger = logging.getLogger(__name__)
+
+
+class MetricReader(ABC):
+    def __init__(
+        self,
+        preferred_temporality: AggregationTemporality = AggregationTemporality.CUMULATIVE,
+    ) -> None:
+        self._collect: Callable[
+            ["MetricReader", AggregationTemporality], Iterable[Metric]
+        ] = None
+        self._preferred_temporality = preferred_temporality
+
+    @final
+    def collect(self) -> None:
+        """Collects the metrics from the internal SDK state and
+        invokes the `_receive_metrics` with the collection.
+        """
+        if self._collect is None:
+            _logger.warning(
+                "Cannot call collect on a MetricReader until it is registered on a MeterProvider"
+            )
+            return
+        self._receive_metrics(self._collect(self, self._preferred_temporality))
+
+    @final
+    def _set_collect_callback(
+        self,
+        func: Callable[
+            ["MetricReader", AggregationTemporality], Iterable[Metric]
+        ],
+    ) -> None:
```

Comment:
not sure on the wording here but feel like it should be called out

```suggestion
    ) -> None:
    """This function is internal to the SDK. It should not be called or override by users"""
```